tests/middlewares/idempotent_tests.rb

```diff
@@ -128,4 +128,34 @@
     response.status
   end
 
+  tests("Retry limit and sleep in constructor with socket erroring first 2 times").returns(200) do
```

Comment:
Testing to ensure this doesn't introduce error seems good, but it seems awkward that it doesn't really change the material (timing) change directly. That said, I'm reticent to try and do that as timing based testing can often be so brittle... Thoughts?